app/utils/helper.ts

```diff
@@ -21,4 +21,25 @@ export const shuffle = <T>(array: T[]): T[] => {
   return array
 }
 
+// Shorten a long text by inserting "..." in the middle, keeping the ends visible.
+export const ellipsizeMiddle = (
+  text: string,
+  options: {
+    maxLength: number
+    maxResultLeft: number
+    maxResultRight: number
+  },
+) => {
+  const { maxLength, maxResultLeft, maxResultRight } = options
+  if (typeof text !== "string") return ""
```

Comment:
this is not necessary we are using typescript and in theory should be detected in compile time